modules/library/metadata/src/main/java/org/geotools/util/Converters.java

```diff
@@ -137,9 +137,7 @@ private static FactoryRegistry getServiceRegistry() {
 
         // handle case of source being an instance of target up front
         final Class<?> sourceClass = source.getClass();
-        if (sourceClass == target
-                || sourceClass.equals(target)
-                || target.isAssignableFrom(sourceClass)) {
+        if (sourceClass == target || target.isAssignableFrom(sourceClass)) {
```

Comment:
I believe the old code was set up as a performance optimization after profiling sessions (``isAssignableFrom`` is an expensive call, not 100% sure but as a native method it might have also synchronization and  CPU register management consequences)

Reply:
Dropped.